backend/tests/mcp_server/test_tool_registry.py

```
@@ -0,0 +1,207 @@
+            assert retrieved_ctx is not None
+            assert retrieved_ctx.token_info.user_id == 3
+            assert retrieved_ctx.tool_name == "test_tool"
+        finally:
+            from app.mcp_server.context import reset_mcp_context
+
+            reset_mcp_context(token)
+
+    def test_context_returns_none_when_not_set(self):
+        """Test that context returns None when not set."""
+        # Clear any existing context
+        ctx = get_mcp_context()
```

Ruff (0.14.14)

[error] Local variable `ctx` is assigned to but never used

Remove assignment to unused variable `ctx`

(F841)